gapic/src/main/com/google/gapid/views/TracerDialog.java

```diff
@@ -597,20 +597,36 @@ private void updateDevicesDropDown(Settings settings) {
         if (device != null && devices != null) {
           deviceLoader.stopLoading();
           device.setInput(devices);
-          if (!settings.traceDevice.isEmpty()) {
-            Optional<DeviceCaptureInfo> deflt = devices.stream()
-                .filter(dev -> settings.traceDevice.equals(dev.device.getSerial()))
-                .findAny();
-            if (deflt.isPresent()) {
-              device.setSelection(new StructuredSelection(deflt.get()));
+          DeviceCaptureInfo deflt = getPreviouslySelectedDevice(settings).orElseGet(() -> {
+            if (devices.size() == 1) {
+              devices.get(0);
+            } else if (devices.size() == 2) {
```

Comment:
This approach will not set any device by default if e.g. 3 devices are listed.

What about:
- returning the first android device found, if any
- otherwise defaulting to the first device listed

e.g.:

```
for (DeviceCaptureInfo dev: devices) {
  if (dev.isAndroid()) {
    return dev
  }
}
if (devices.size() > 0) {
  // default to first listed device
  return devices.get(0);
}
return null;
```

Reply:
No, If there are more than one Android device, selecting the wrong one is no better than selecting nothing. The reality is that 1) more than one device is uncommon, and 2) selecting the previous selected device will happen 99% of the time.

Also, device come in arbitrary order, so selecting "the first one" is actually meaningless.

Reply:
Personally, when no previously selected devices are present, I would prefer the UI to pick up any device rather than showing a blank field. But I'm OK with your current implem also.